internal/multigitter/shared.go

```diff
@@ -70,3 +73,73 @@ func ParseConflictStrategy(str string) (ConflictStrategy, error) {
 		return ConflictStrategyReplace, nil
 	}
 }
+
+// CreateTempDir creates a temporary directory in the given directory.
+// If the given directory is an empty string, it will use the os.TempDir()
+func CreateTempDir(cloneDir string) (string, error) {
+	if cloneDir == "" {
+		cloneDir = os.TempDir()
+	}
+
+	absDir, err := createAbsolutePath(cloneDir)
+	if err != nil {
+		return "", err
+	}
+
+	err = createDirectoryIfDoesntExist(absDir)
+	if err != nil {
+		return "", err
+	}
+
+	tmpDir, err := os.MkdirTemp(absDir, "multi-git-changer-")
+	if err != nil {
+		return "", err
+	}
+
+	return tmpDir, nil
+}
+
+func createDirectoryIfDoesntExist(directoryPath string) error {
+	// Check if the directory exists
+	if _, err := os.Stat(directoryPath); os.IsNotExist(err) {
+		// Create the directory
+		err := os.MkdirAll(directoryPath, os.ModePerm)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return nil
+}
+
+func createAbsolutePath(path string) (string, error) {
+	// Handle empty path
+	if path == "" {
+		return os.TempDir(), nil
+	}
```

Comment:
I think this check will give an unexpected result. "Gettings an absolute path" should not return the temporary directory. 

Either remove the check altogether since we only call this from `CreateTempDir`, or make it error.